slurmutils/models/option.py

```diff
@@ -0,0 +1,449 @@
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Configuration options for Slurm data models."""
+
+__all__ = [
+    "SlurmdbdConfigOptions",
+    "SlurmConfigOptions",
+    "NodeOptions",
+    "DownNodeOptions",
+    "FrontendNodeOptions",
+    "NodeSetOptions",
+    "PartitionOptions",
+]
+
+from dataclasses import dataclass, fields
+from typing import Iterable
+
+from .callback import (
+    Callback,
+    ColonSeparatorCallback,
+    CommaSeparatorCallback,
+    ReasonCallback,
+    SlurmDictCallback,
+)
+
+
+@dataclass(frozen=True)
+class _Option:
```

Comment:
Nit: seems weird to call this `_Option` when it's the base class for a list of options, so maybe something like `_OptionsList` could be better?

Reply:
How about `_OptionSet`?

I originally had the name to be `_OptionList`, but I changed it to just `_Option` because "List" implies to me that it's an ordered data structure when really it's just a set of valid options for various configuration objects with callbacks for certain configuration option values.

Reply:
Yeah, `_OptionSet` sounds good!

Reply:
Implemented `_OptionSet` and better naming overall for the configuration option sets in https://github.com/charmed-hpc/slurmutils/pull/15/commits/703303ff30e6e3631c9c0320376670d33ddcf257